tests/sdk/context/skill/test_skill_serialization.py

```diff
@@ -0,0 +1,313 @@
+    serialized = task_skill.model_dump()
+    assert serialized["trigger"]["type"] == "task"
+    assert serialized["name"] == "test-task"
+    assert "Task-based instructions with ${variable}" in serialized["content"]
+    assert serialized["trigger"]["triggers"] == ["task", "automation"]
+    assert len(serialized["inputs"]) == 1
+    assert serialized["inputs"][0]["name"] == "variable"
+
+    # Test JSON serialization
+    json_str = task_skill.model_dump_json()
+    assert isinstance(json_str, str)
+    parsed = json.loads(json_str)
+    assert parsed["trigger"]["type"] == "task"
+
+    # Test deserialization
+    deserialized = Skill.model_validate(serialized)
+    assert isinstance(deserialized.trigger, TaskTrigger)
+    assert deserialized.name == "test-task"
+    assert deserialized.trigger.triggers == ["task", "automation"]
+    assert len(deserialized.inputs) == 1
+
+
+def test_skill_union_serialization_roundtrip():
+    """Test complete serialization roundtrip for all trigger types."""
+    # Test data for each trigger type
+    test_cases = [
+        Skill(
+            name="repo-test",
+            content="Repo content",
+            source="repo.md",
+            trigger=None,
+        ),
+        Skill(
+            name="knowledge-test",
+            content="Knowledge content",
+            source="knowledge.md",
+            trigger=KeywordTrigger(keywords=["test"]),
+        ),
+        Skill(
+            name="task-test",
+            content="Task content with ${var}",
+            source="task.md",
+            trigger=TaskTrigger(triggers=["task"]),
+            inputs=[InputMetadata(name="var", description="Test variable")],
+        ),
+    ]
+
+    for original_skill in test_cases:
+        # Serialize to dict
+        serialized = original_skill.model_dump()
+
+        # Serialize to JSON string
+        json_str = original_skill.model_dump_json()
+
+        # Deserialize from dict
+        deserialized_from_dict = Skill.model_validate(serialized)
+
+        # Deserialize from JSON string
+        deserialized_from_json = Skill.model_validate_json(json_str)
+
+        # Verify all versions are equivalent
+        # Handle None trigger (always-active skills)
+        if original_skill.trigger is None:
+            assert deserialized_from_dict.trigger is None
+            assert deserialized_from_json.trigger is None
+        else:
+            assert deserialized_from_dict.trigger.type == original_skill.trigger.type
+            assert deserialized_from_json.trigger.type == original_skill.trigger.type
+        assert deserialized_from_dict.name == original_skill.name
+        assert deserialized_from_dict.content == original_skill.content
+        assert deserialized_from_dict.source == original_skill.source
+        assert deserialized_from_json.name == original_skill.name
+        assert deserialized_from_json.content == original_skill.content
+        assert deserialized_from_json.source == original_skill.source
```

Comment:
We should just `assert deserialized == original` instead of trying to check all attributes